pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -49,9 +49,9 @@ var (
 	monitorSecret   = "ROOK_MON_SECRET"
 )
 
-func (c *ClusterController) startClusterCleanUp(cluster *cephv1.CephCluster, cephHosts []string, monSecret string) {
+func (c *ClusterController) startClusterCleanUp(stopCleanupCh chan struct{}, cluster *cephv1.CephCluster, cephHosts []string, monSecret string) {
```

Comment:
What about `stopCleanupCh` as part of `ClusterController`?

Reply:
We need a new channel with each reconcile. When the reconcile is re-queued, we need to close the channel immediately. We can't use the same channel across multiple reconciles to stop the cleanup goroutine.